28_FFTBloom/main.cpp

```diff
@@ -118,88 +109,102 @@ class FFTBloomApp final : public examples::SimpleWindowedApplication, public app
 
 	// -------------------------------- END WINDOWED APP OVERRIDES ---------------------------------------------------
 
-	smart_refctd_ptr<IGPUPipelineLayout> createPipelineLayout(const std::span<const IGPUDescriptorSetLayout::SBinding> bindings)
-	{
-		const asset::SPushConstantRange pcRange = { .stageFlags = IShader::E_SHADER_STAGE::ESS_COMPUTE,.offset = 0,.size = sizeof(PushConstantData) };
-		return m_device->createPipelineLayout({ &pcRange,1 }, m_device->createDescriptorSetLayout(bindings));
-	}
-
-	inline void updateDescriptorSet(smart_refctd_ptr<IGPUImageView> imageDescriptor, smart_refctd_ptr<IGPUSampler> samplerDescriptor, smart_refctd_ptr<IGPUImageView> storageImageDescriptor, smart_refctd_ptr<IGPUImageView> textureArrayDescriptor = nullptr)
+	inline void updateDescriptorSet(smart_refctd_ptr<IGPUImageView> imageDescriptor, smart_refctd_ptr<IGPUImageView> storageImageDescriptor, smart_refctd_ptr<IGPUImageView> textureArrayDescriptor = nullptr)
 	{
-		IGPUDescriptorSet::SDescriptorInfo infos[4] = {};
-		IGPUDescriptorSet::SWriteDescriptorSet writes[4] = {};
+		IGPUDescriptorSet::SDescriptorInfo infos[3] = {};
+		IGPUDescriptorSet::SWriteDescriptorSet writes[3] = {};
 
-		for (auto i = 0u; i < 4; i++) {
+		for (auto i = 0u; i < 3; i++) {
 			writes[i].dstSet = m_descriptorSet.get();
-			writes[i].binding = i;
 			writes[i].arrayElement = 0u;
 			writes[i].count = 1u;
 			writes[i].info = &infos[i];
 		}
 
 		infos[0].desc = imageDescriptor;
 		infos[0].info.image.imageLayout = IImage::LAYOUT::READ_ONLY_OPTIMAL;
+		// Read image at binding 0
+		writes[0].binding = 0u;
 
-		infos[1].desc = samplerDescriptor;
+		// Binding 2 skipped since it's the sampler which we never need to change
 
-		infos[2].desc = storageImageDescriptor;
-		infos[2].info.image.imageLayout = IImage::LAYOUT::GENERAL;
+		infos[1].desc = storageImageDescriptor;
+		infos[1].info.image.imageLayout = IImage::LAYOUT::GENERAL;
+		// Storage image at binding 1
+		writes[1].binding = 2u;
 
 		// If nullptr give it SOME value so validation layer doesn't complain even though we don't use it
-		infos[3].desc = textureArrayDescriptor ? textureArrayDescriptor : imageDescriptor;
-		infos[3].info.image.imageLayout = IImage::LAYOUT::READ_ONLY_OPTIMAL;
+		infos[2].desc = textureArrayDescriptor ? textureArrayDescriptor : imageDescriptor;
+		infos[2].info.image.imageLayout = IImage::LAYOUT::READ_ONLY_OPTIMAL;
+		// Texture array for reading at binding 3
+		writes[2].binding = 3u;
 
-		m_device->updateDescriptorSets({ writes, 4 }, std::span<IGPUDescriptorSet::SCopyDescriptorSet>());
+		m_device->updateDescriptorSets(writes, std::span<IGPUDescriptorSet::SCopyDescriptorSet>());
 	}
```

Comment:
if you fix https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/144/files/dde8eb59280c409268fbf1ee8a7b825338494af5..513755c14f0af1f9e0bef1529116229fd9ad9298#r1892208818, do you need this function anymore ?

Reply:
I guess I could make the kernel spectra be `LAYOUT::GENERAL` and never have to touch anything once upload is done

Reply:
End up needing it regardless because I use the same DS for kernel FFT and image convolution so most things bound for the former change for the latter